src/main/scala/rtr/Retry.scala

```diff
@@ -0,0 +1,31 @@
+package rtr
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+object Retry extends App {
+
+  @tailrec
+  def retry[A](block: () => A, acceptResult: A => Boolean, retries: List[FiniteDuration]): A = {
+
+    val a = block()
+    if (acceptResult(a)) {
+      a
+    } else {
+      retries match {
+        case Nil => a
+        case x :: xs => {
+          Thread.sleep(x.toMillis)
+          retry(block, acceptResult, xs)
+        }
+      }
+    }
+  }
+
+  def retryf[A](block: () => Future[A], acceptResult: A => Boolean, retries: List[FiniteDuration]): Future[A] = {
+   block().map(a=>retry(()=>a,acceptResult,retries))
```

Comment:
1. retryF should call itself and be recursive.
2. your version will evaluate block() result **only once** and pass it to retry. Imagine you pass to retryf a httpcall function: in this case you will make only one http request which is not what really we need, we need to evaluate the function at each retry step.